src/stream_pipe.cc

```diff
@@ -261,14 +239,46 @@ void StreamPipe::WritableListener::OnStreamRead(ssize_t nread,
   return previous_listener_->OnStreamRead(nread, buf);
 }
 
+Maybe<StreamPipe*> StreamPipe::New(StreamBase* source,
```

Comment:
Is this a new pattern we’re following? `Maybe<Type*>`? I’ve been a bit out of the loop but in all cases that I was familiar with we just used the raw pointer type since pointers are already `Maybe<>`-ish